### PR TITLE
Improve copying from DataView and CallTreeWidget, in particular with spreadsheets in mind

### PR DESCRIPTION
**Pull request by dpallotti**

### CODE REVIEW

**Comment by dpallotti** [comment]
I guess that the debatable part is only copying percentages for columns that visualize percentage+number. I think that adding columns instead is quite messy, both for `DataView`s and for `CallTreeWidget`, so I decided not to do that. After all, until not long ago the `SamplingReportDataView` was only showing percentages. Let me know what you think.

File: src/OrbitQt/CallTreeWidget.cpp [hunk 1]
```diff
@@ -235,17 +235,20 @@ static std::string BuildStringFromIndices(QTreeView* tree_view, const QModelInde
...
+      }
 
       switch (item.expanded_state) {
         case ItemWithAncestorRows::kLeaf:
-          buffer.append("  ");
+          buffer.append(kNoBreakSpace).append(kNoBreakSpace);
           break;
         case ItemWithAncestorRows::kExpanded:
-          buffer.append("- ");
+          buffer.append("▾").append(kNoBreakSpace);
```
Comment (beckerhe) [inline src/OrbitQt/CallTreeWidget.cpp:279]: Nice!

**Review by beckerhe — APPROVED** [review]
Looks all good to me!

File: src/OrbitQt/CallTreeWidget.cpp [hunk 2]
```diff
@@ -120,8 +120,26 @@ class HideValuesForBottomUpProxyModel : public QIdentityProxyModel {
         proxy_index.column() == CallTreeViewItemModel::kOfParent) {
       return QVariant{};
     }
+
+    // The "Exclusive" column is not shown in the UI for the bottom-up tree, as its values wouldn't
+    // be useful. When copying, leave it empty instead.
+    if (role == CallTreeViewItemModel::kCopyableValueRole &&
+        proxy_index.column() == CallTreeViewItemModel::kExclusive) {
+      return QVariant{};
+    }
```
[the remark this answers is not in the document]
Comment (dpallotti) [reply src/OrbitQt/CallTreeWidget.cpp:129]: The consistency would only be with the top-down view, but yes, that's a point in favor.
But the main reason was probably simplicity. Basically I'm not sure what the least hacky way would be to hide the column only for the bottom-up view when copying. For displaying, `hideColumn` is used. Should I add another user role and query `headerData`? Should I add a `copy_exclusive_` field to `CallTreeWidget`? Or should I just leave the column empty as it's done now?

[the remark this answers is not in the document]
Comment (beckerhe) [reply src/OrbitQt/CallTreeWidget.cpp:129]: I guess the proper solution is to dynamically output only the columns that are acutally shown in the widget.

But I'm fine with leaving it like this "for consistency reasons". :sweat_smile:

[the remark this answers is not in the document]
Comment (dpallotti) [reply src/OrbitQt/CallTreeWidget.cpp:129]: Aaah, there's an `isColumnHidden`, I didn't think about it. I'll take a look...

[the remark this answers is not in the document]
Comment (dpallotti) [reply src/OrbitQt/CallTreeWidget.cpp]: Done. The result is better now. I rewrote the last commit.

**Pull request merged**